pebblo/app/service/discovery/common.py

```diff
@@ -0,0 +1,48 @@
+from pebblo.app.enums.enums import ApplicationTypes
+from pebblo.app.utils.utils import timeit
+from pebblo.log import get_logger
+
+logger = get_logger(__name__)
+
+
+@timeit
+def get_or_create_app(db, app_name, app_class, data, app_type):
+    """
+    Gets or creates an AiApp
+    """
+    try:
+        logger.debug(f"Fetching or creating {app_class.__tablename__} details")
+        exist, ai_app = db.query(app_class, {"name": app_name})
+        if exist and ai_app and len(ai_app) > 0:
+            logger.debug(f"Application details exists in {app_class.__tablename__}")
+            return ai_app[0]
+
+        ai_app = {"name": app_name}
+
+        if app_type == ApplicationTypes.LOADER.value:
+            ai_app["id"] = data["load_id"]
+        elif app_type == ApplicationTypes.RETRIEVAL.value:
+            pass
+
+        exist, existing_ai_app = db.query(app_class, ai_app)
+        if exist and existing_ai_app:
+            logger.debug(f"Application details exists in {app_class.__tablename__}")
```

Comment:
We should add a specific prefix in the logger message like - LoaderApp:  as we had done in last release